### DISCUSSION TITLE
Will SSD size start to shrink when worn out?

### DISCUSSION THREAD

**Discussion by pluto2021** [Chia-Network/chia-blockchain#2431] [Q&A]
When the available spares run out, would the flash memory controller starts to mark bad bits or that won’t be the case and the SSD drive would become faulty with bad bits not corrected, leading to corrupted operations and has to be discarded?

**Comment by HaitaoMiao** [comment]
Yes, You are right, Bad blocks will increase.

**Reply by pluto2021** [reply]
Do you know of any technical references? If I were to design the SSD drive algorithm, I would make it usable as long as it can hold data by marking out bad bits and replace with other good bits, essentially at the expense of the drive capacity. It is like making a car that is still safe to drive despite of a worn engine. Imagine if car engines would simply quit working due to a necessary engine oil, it would be insane. I would guess that this would essentially differentiate the more reliable SSDs comparing with those that were made just to work with light usage.
